### PR TITLE
Feature/open lease request

### PR DESCRIPTION
**Pull request by Sagie501**

### CODE REVIEW

File: src/app/core/guards/checkout-guard/checkout.guard.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return this.productsService.getProductById(next.params.id).pipe(
+      switchMap((product) => {
+        if (product) {
+          return this.userStore.select(getLoggedInUser).pipe(
+            switchMap((loggedInUser) => {
+              if (loggedInUser) {
+                return this.productsService.getProductsByUserId(loggedInUser.id).pipe(
+                  map((userProducts) => {
+                    if (userProducts.find((product) => product.id === next.params.id)) {
```
Comment (tomshudev) [inline src/app/core/guards/checkout-guard/checkout.guard.ts:28]: What are you checking here?
Is it checking whether the user is trying to lend its own product?

File: src/app/features/checkout/components/credit-card-simulator/credit-card-simulator.component.html [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
-    *ngIf="isVisa"
-    src="http://pngimg.com/uploads/visa/visa_PNG36.png"
-    alt="Visa logo"
-  />
-  <img
-    class="image mastercard"
-    *ngIf="!isVisa"
-    src="http://pngimg.com/uploads/mastercard/mastercard_PNG9.png"
-    alt="MasterCard logo"
+    class="image {{creditCardType}}"
```
Comment (tomshudev) [inline src/app/features/checkout/components/credit-card-simulator/credit-card-simulator.component.html:4]: The BEST!
Great change

File: src/app/features/checkout/components/payment/payment.component.html [hunk 3]
```diff
@@ -2,19 +2,17 @@
   <div class="headline">Card Details</div>
 
   <ile-credit-card-simulator
-    class="credit-card"
+    class="credit-card card-right"
```
Comment (tomshudev) [inline src/app/features/checkout/components/payment/payment.component.html:5]: Should it be on the left, right or center?
What looks better?

File: src/app/features/checkout/components/payment/payment.component.ts [hunk 4]
```diff
@@ -64,4 +84,12 @@ export class PaymentComponent implements OnInit {
     }
     return '';
   }
+
+  checkout() {
+    console.log(this.paymentForm.value);
```
Comment (tomshudev) [inline src/app/features/checkout/components/payment/payment.component.ts:89]: Can't we open the lease request here?
We have all the queries, we only need to send a request, am I wrong?

File: src/app/features/product-page/containers/product-page/product-page.component.html [hunk 5]
```diff
@@ -30,37 +30,29 @@
...
-      />
-      <label
-        *ngFor="
-          let value of [5, 4.5, 4, 3.5, 3, 2.5, 2, 1.5, 1, 0.5];
-          let i = index
-        "
-        for="star-{{ value }}"
-        [ngClass]="{ full: i % 2 === 0, half: i % 2 !== 0 }"
-      ></label>
+    <fieldset class="rating" [ngStyle]="{'pointer-events': isRated ? 'none' : 'auto'}">
```
Comment (tomshudev) [inline src/app/features/product-page/containers/product-page/product-page.component.html:33]: Why did you gave up on the for loop?

File: src/app/features/user/my-products/components/product-dialog/product-dialog.component.html [hunk 6]
```diff
@@ -41,7 +41,8 @@
 
       <mat-form-field>
         <mat-label>Product images</mat-label>
-        <ngx-mat-file-input formControlName="imageFiles" required multiple accept="image/*"></ngx-mat-file-input>
+        <ngx-mat-file-input formControlName="imageFiles" required multiple accept="image/*"
+                            [errorStateMatcher]="matcher"></ngx-mat-file-input>
```
Comment (tomshudev) [inline src/app/features/user/my-products/components/product-dialog/product-dialog.component.html:45]: What is it doing?

File: src/app/core/guards/checkout-guard/checkout.guard.ts (hunk 1 above)
Comment (Sagie501) [reply src/app/core/guards/checkout-guard/checkout.guard.ts:28]: First I check if the product with the given id really exist. If it is exist, yes, I check if the use try to lend his own product.

File: src/app/features/checkout/components/payment/payment.component.html (hunk 3 above)
Comment (Sagie501) [reply src/app/features/checkout/components/payment/payment.component.html:5]: You are the one with the better taste between us..

File: src/app/features/checkout/components/payment/payment.component.ts (hunk 4 above)
Comment (Sagie501) [reply src/app/features/checkout/components/payment/payment.component.ts:89]: Yes this is the next thing I am going to do

File: src/app/features/product-page/containers/product-page/product-page.component.html (hunk 5 above)
Comment (Sagie501) [reply src/app/features/product-page/containers/product-page/product-page.component.html:33]: I thought that don't break the styles but it did.. So I just revert it..

File: src/app/features/user/my-products/components/product-dialog/product-dialog.component.html (hunk 6 above)
Comment (Sagie501) [reply src/app/features/user/my-products/components/product-dialog/product-dialog.component.html:45]: It is control when error messages are shown. I put it in the whole form but forgot this one

File: src/app/core/guards/checkout-guard/checkout.guard.ts (hunk 1 above)
Comment (tomshudev) [reply src/app/core/guards/checkout-guard/checkout.guard.ts:28]: Got it

**Pull request merged**